tests/src/controller/avdeccControlledEntity_tests.cpp

```diff
@@ -113,6 +113,29 @@ TEST(ControlledEntity, AddChannelMappings)
 	EXPECT_TRUE(Mapping == e.getStreamPortInputAudioMappings(StreamPort).at(0));
 }
 
+TEST(ControlledEntity, GetInvalidMappings)
+{
+	auto const flags = la::avdecc::entity::model::jsonSerializer::Flags{ la::avdecc::entity::model::jsonSerializer::Flag::IgnoreAEMSanityChecks, la::avdecc::entity::model::jsonSerializer::Flag::ProcessADP, la::avdecc::entity::model::jsonSerializer::Flag::ProcessCompatibility, la::avdecc::entity::model::jsonSerializer::Flag::ProcessDynamicModel, la::avdecc::entity::model::jsonSerializer::Flag::ProcessMilan, la::avdecc::entity::model::jsonSerializer::Flag::ProcessState, la::avdecc::entity::model::jsonSerializer::Flag::ProcessStaticModel, la::avdecc::entity::model::jsonSerializer::Flag::ProcessStatistics };
+	auto controller = la::avdecc::controller::Controller::create(la::avdecc::protocol::ProtocolInterface::Type::Virtual, "VirtualInterface", 0x0001, la::avdecc::UniqueIdentifier{}, "en");
+	auto const [error, message] = controller->loadVirtualEntityFromJson("data/Listener_EmptyMappings.json", flags);
+	EXPECT_EQ(la::avdecc::jsonSerializer::DeserializationError::NoError, error);
+	EXPECT_STREQ("", message.c_str());
+
+	auto& e = const_cast<la::avdecc::controller::ControlledEntityImpl&>(static_cast<la::avdecc::controller::ControlledEntityImpl const&>(*controller->getControlledEntityGuard(la::avdecc::UniqueIdentifier{ 0x001B92FFFF000001 })));
+	constexpr auto StreamPort = la::avdecc::entity::model::StreamPortIndex{ 1u };
+
+	// Add mapping to 2nd stream channel
+	auto const Mapping = la::avdecc::entity::model::AudioMapping{ 0, 1, 0, 0 };
+	e.addStreamPortInputAudioMappings(StreamPort, la::avdecc::entity::model::AudioMappings{ Mapping });
+
+	// Calculate invalid mappings for a stream format with a single channel (AAF 48kHz 1ch)
+	auto const StreamFormat = la::avdecc::entity::model::StreamFormat(0x0205022000406000);
+	auto const invalid = e.getStreamPortInputInvalidAudioMappingsForStreamFormat(0, StreamFormat);
+	EXPECT_EQ(1, invalid.size()) << "Exactly one stream port should have invalid mappings";
```

Comment:
I don't think this is actually sufficient. The `map` might still contain a different stream port, causing `.at` to throw.
I'll keep this, so the test keeps running for sizes > 1, but add an assert for the key existing.

Reply:
You are correct this is not enough, but (new) tests should only try to test one thing at a time, that's why I was suggestion to change this line to at least stop the test here in case the source is changed for some reason (eg. if someone adds more stream ports, like you did to the existing data source, it alter even more the test and not gives the expected test result).
I think it's more safe to either use a data source per test (for specific feature test), and/or to include assertions to the test to protect the test case from external changes (in data source).

So in this regard, I would change to ASSERT_EQ and include testing for the correct StreamPort, if that's ok for you.

Reply:
Oh yeah you are right, I forgot you manually added mappings in the test itself :) But in any case, I think it's better to either end the test right away if test requierements are not fulfilled, or protect the test from crashing.
Anyway, it should be ok, thanks a lot for taking the time to add the test.